lib/rules/sort-keys.js

```diff
@@ -156,13 +177,24 @@ module.exports = {
                 }
 
                 const prevName = stack.prevName;
+                const prevEndLine = stack.prevEndLine;
                 const numKeys = stack.numKeys;
                 const thisName = getPropertyName(node);
+                const thisStartLine = node.loc.start.line;
+                const isLineSeparatedGroups = prevEndLine && checkLineSeparatedGroups(thisStartLine, prevEndLine);
+
+                if (thisStartLine !== null) {
```

Comment:
I think it's unneeded condition since `thisStartLine` seems equals with `node.loc.start.line` (`number`).